test/stub-test.js

```diff
@@ -285,6 +285,65 @@ describe("stub", function () {
         });
     });
 
+    describe(".resolvesThis", function () {
+        afterEach(function () {
+            if (Promise.resolve.restore) {
+                Promise.resolve.restore();
+            }
+        });
+
+        it("returns a promise resolves with this", function () {
+            var instance = {};
+            instance.stub = createStub.create();
+            instance.stub.resolvesThis();
+
+            return instance.stub().then(function (actual) {
+                assert.same(actual, instance);
+            });
+        });
+
+        var strictMode = (function () {
+            return this;
+        }()) === undefined;
+        if (strictMode) {
```

Comment:
I am not sure a conditional for strict mode is necessary:

* Sinon's source is meant to be ES5.1, strict mode and expects to run in engines that supports that.
* The test file has `use strict;` at the top.

If for some reason, someone decides to run the tests in non-strict mode or in engines that doesn't support that, then I think the tests **should** be allowed to fail.